work_queue/src/work_queue.c

```diff
@@ -893,8 +934,9 @@ void update_read_catalog_factory(struct work_queue *q, time_t stoptime) {
 
 	hash_table_firstkey(q->factory_table);
 	while ( hash_table_nextkey(q->factory_table, factory_name, (void **) f) ) {
```

Comment:
Check other uses of `hash_table_nextkey`. You are missing the & in both key and value.